src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,83 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int bucketIndex = hash(key);
+        Node<K, V> node = getNode(bucketIndex, key);
+        if (node == null) {
+            table[bucketIndex] = new Node(key, value, table[bucketIndex]);
+            size++;
+        } else {
+            node.value = value;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = getNode(hash(key), key);
+        return node == null ? null : node.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
```

Comment:
What value does this method return? Hash code?

Reply:
@dmytro-hryhoruk yes. I named it as "hash()" and not as "getHashCode()" because method that returns "key.hashCode() % table.length" named hash() in default implementation